tests/unit/types/request/test_request.py

```diff
@@ -116,4 +118,26 @@ def test_status():
     deserialized_request = DataRequestProto.FromString(byte_array)
     assert not deserialized_request.is_decompressed
     assert deserialized_request.status.code == jina_pb2.StatusProto.ERROR
+    assert deserialized_request.is_decompressed_wo_data
+    assert deserialized_request.is_decompressed
+
+
+def test_lazy_parameters():
+    doc_count = 1000
+    r = DataRequest()
+    da = r.docs
+    da.extend([Document(text='534534534er5yr5y645745675675675345')] * doc_count)
+    r.data.docs = da
+
+    parameters = {'a': 0}
+    r.parameters = parameters
+    byte_array = DataRequestProto.SerializeToString(r)
+
+    deserialized_request = DataRequestProto.FromString(byte_array)
```

Comment:
This should be callig the regular DataRequest contstructor, this is a unit test independent of the Proto optimization we have in serializer